test/inspector/inspector-helper.js

```diff
@@ -0,0 +1,412 @@
+'use strict';
+const assert = require('assert');
+const common = require('../common');
+const fs = require('fs');
+const http = require('http');
+const path = require('path');
+const spawn = require('child_process').spawn;
+
+const TIMEOUT = 15 * 1000;
+
+const mainScript = path.join(
+  common.fixturesDir,
+  'loop.js'
+);
+
+function send(socket, message, id, callback) {
+  const msg = JSON.parse(JSON.stringify(message)); // Clone!
+  msg['id'] = id;
+  const message_buf = Buffer.from(JSON.stringify(msg));
+
+  const ws_header_buf = Buffer.allocUnsafe(16);
+  ws_header_buf.writeUInt8(0x81, 0);
+  let byte2 = 0x80;
+  const body_len = message_buf.length;
+  let mask_offset = 2;
+  if (body_len < 126) {
+    byte2 = 0x80 + body_len;
+  } else if (body_len < 65536) {
+    byte2 = 0xFE;
+    ws_header_buf.writeUInt16BE(body_len, 2);
+    mask_offset = 4;
+  } else {
+    byte2 = 0xFF;
+    ws_header_buf.writeUInt32BE(body_len, 2);
+    ws_header_buf.writeUInt32BE(0, 6);
+    mask_offset = 10;
+  }
+  ws_header_buf.writeUInt8(byte2, 1);
+  ws_header_buf.writeUInt32BE(0x01020408, mask_offset);
+
+  for (let i = 0; i < message_buf.length; i++)
+    message_buf[i] = message_buf[i] ^ (1 << (i % 4));
+  socket.write(
+      Buffer.concat([ws_header_buf.slice(0, mask_offset + 4), message_buf]),
+      callback);
+}
+
+function processIncomingMessage(socket, buffer, handler) {
+  if (buffer.length < 2)
+    return 0;
+  assert.equal(0x81, buffer[0]);
+  let dataLen = 0x7F & buffer[1];
+  let bodyOffset = 2;
+  if (buffer.length < bodyOffset + dataLen)
+    return 0;
+  if (dataLen === 126) {
+    dataLen = buffer.readUInt16BE(2);
+    bodyOffset = 4;
+  } else if (dataLen === 127) {
+    dataLen = buffer.readUInt32BE(2);
+    bodyOffset = 10;
+  }
+  if (buffer.length < bodyOffset + dataLen)
+    return 0;
+  const message = JSON.parse(
+      buffer.slice(bodyOffset, bodyOffset + dataLen).toString('utf8'));
+  handler(message);
+  return bodyOffset + dataLen;
+}
+
+function tearDown(child, err) {
+  child && child.kill();
```

Comment:
Is it necessary to check if the child exists? Shouldn't we be able to guarantee that in our tests?